torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -514,50 +549,65 @@ GraphCache::GraphCache(std::shared_ptr<Graph> graph)
   // compile a kernel if we have enough information from graph (profiling
   // record)
   if (IsNewExecutorEnabled()) {
-    createFusionExecutorCache(
+    appendFusionExecutorCache(
         InputsRequirement(graph_, toVector(reduction_axes_)));
   }
 }
 
 std::vector<at::Tensor> GraphCache::runGraphWithInputs(
     const at::ArrayRef<IValue>& inputs) {
-  InputsRequirement input_stack(inputs, toVector(reduction_axes_));
+  // get unique id `unique_id` for given input set `inputs`;
+  const size_t unique_id = inputs_id_lookup_.getCode(inputs);
+
   FusionExecutorCache* fusion_executor_cache = nullptr;
 
-  // TODO: hash indexing;
-  for (size_t i = 0; i < fe_cache_.size(); i++) {
-    if (input_stack.complyWith(input_stacks_[i])) {
-      fusion_executor_cache = fe_cache_[i].get();
-      break;
+  if (code_to_index_lookup_.count(unique_id) == 0) {
+    InputsRequirement input_stack(inputs, toVector(reduction_axes_));
+    for (size_t i = 0; i < fe_cache_.size(); i++) {
+      if (input_stack.complyWith(input_stacks_[i])) {
+        // found compliable fe_cache_ entry
+        fusion_executor_cache = fe_cache_[i].get();
+        // record short cut to designated fusion executor
+        code_to_index_lookup_[unique_id] = i;
+        break;
+      }
     }
+    if (!fusion_executor_cache) {
+      fusion_executor_cache = appendFusionExecutorCache(input_stack);
```

Comment:
Does this mean you're creating a cache entry before we enter into runFusion?

Reply:
The short answer is no.

This is the ugly bit, each level of cache has their own entry. At this point, we are creating an instance of `FusionExecutorCache` as well as a cache entry for `GraphCache` (the hosting cache).

But we are not creating any cache entry for nested structures. We only create cache entry below when we later call `fusion_executor_cache->runFusionWithInputs`